dacite.py

```diff
@@ -248,7 +251,7 @@ def _inner_from_dict_for_union(data: Any, field: Field, outer_config: Config) ->
                     outer_config=outer_config,
                     field=field,
                 )
-            elif _is_instance(t, data):
+            elif _is_instance(t, data) or not outer_config.validate_types:
```

Comment:
Does it mean that we give up after first try? Maybe we should do something like that at the end of this function:

```python
if not outer_config.validate_type:
    return data
raise UnionMatchError(field)
```